app/src/main/kotlin/com/wire/android/ui/migration/MigrationViewModel.kt

```diff
@@ -31,53 +32,77 @@ import com.wire.android.navigation.BackStackMode
 import com.wire.android.navigation.NavigationCommand
 import com.wire.android.navigation.NavigationItem
 import com.wire.android.navigation.NavigationManager
-import com.wire.android.util.dispatchers.DispatcherProvider
 import com.wire.android.workmanager.worker.enqueueMigrationWorker
+import com.wire.android.workmanager.worker.enqueueSingleUserMigrationWorker
+import com.wire.kalium.logic.data.user.UserId
 import com.wire.kalium.logic.feature.session.CurrentSessionResult
 import com.wire.kalium.logic.feature.session.CurrentSessionUseCase
 import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.launch
 import javax.inject.Inject
+import com.wire.android.appLogger
+import com.wire.android.navigation.EXTRA_USER_ID
+import com.wire.kalium.logger.obfuscateId
+import com.wire.kalium.logic.data.id.QualifiedIdMapperImpl
 
 @HiltViewModel
 class MigrationViewModel @Inject constructor(
     private val navigationManager: NavigationManager,
     private val getCurrentSession: CurrentSessionUseCase,
     private val workManager: WorkManager,
-    private val dispatchers: DispatcherProvider,
+    private val savedStateHandle: SavedStateHandle
 ) : ViewModel() {
 
     var state: MigrationState by mutableStateOf(MigrationState.InProgress(MigrationData.Progress.Type.UNKNOWN))
         private set
 
     init {
-        viewModelScope.launch(dispatchers.io()) {
-            enqueueMigrationAndListenForStateChanges()
+        viewModelScope.launch {
+            enqueueMigrationAndListenForStateChanges(getUserIdFromNavigationArgs())
         }
     }
 
+    private fun getUserIdFromNavigationArgs(): UserId? =
```

Comment:
Maybe it would be better to make it a `val` instead of `fun`?
Or even create a sealed class `MigrationType` with two types `AllAccounts` and `SingleAccount(userId)`.
What do you think?